account_invoice_ubl_peppol/README.rst

```diff
@@ -0,0 +1,122 @@
+   !! This file is generated by oca-gen-addon-readme !!
+   !! changes will be overwritten.                   !!
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+.. |badge1| image:: https://img.shields.io/badge/maturity-Beta-yellow.png
+    :target: https://odoo-community.org/page/development-status
+    :alt: Beta
+.. |badge2| image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fedi-lightgray.png?logo=github
+    :target: https://github.com/OCA/edi/tree/13.0/account_invoice_ubl_peppol
+    :alt: OCA/edi
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/edi-13-0/edi-13-0-account_invoice_ubl_peppol
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/226/13.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+With module `Account Invoice UBL`, invoices are generated according to generic UBL rules.
+
+In Europe, some countries use the PEPPOL 3.0 BIS standard as a more strict subset of UBL.
+
+With this module you can specify some or all of your invoices to be generated and validated
+according to this stricter standard.
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Configuration
+=============
+
+- Go to menu *Invoicing > Configuration > Settings > Invoicing*, under *Electronic Invoices*.
+
+- Formulate a domain for which invoices the dialect should become PEPPOL.
+  By default it is *[]*, so all UBL invoices will be PEPPOL.
+  If you want this only for Belgian partners, then you can fill here for example:
+  `[('partner_id.country_id.code', '=', 'BE')]`
+  Or you can choose to enable this only for selected partner ids.
+
+- You can configure a default tax to use in case an invoice line has no tax specified.
+  This is necessary for example in case of an NGO to satisfy business rule
+  BR-CO-18. Any tax you choose must also have a UNECE tax type (eg. VAT) and tax
+  category (eg. "Services outside scope of tax") defined.
+
+- You can configure a default unit of measure, of which the UNECE code will be used
+  in case an invoice line has no unit or product unit. A typical default unit could
+  be the Odoo 'unit', configured with a UNECE code of UN, XUN or C62. This is to
+  satisfy rule BR-23.
+
+- Go to menu *Contacts*
+  Fill the field `coc_registration_number` for your own company's partner record and for
+  those partners that you want to send e-invoices to.
+
+- Go to menu *Contacts > Configuration > Localization > Countries*
+  On any country relevant for invoice traffic, configure the correct PEPPOL EAS id.
+  For the Netherlands, this is for example `0106`, which stands for Dutch chamber of
+  commerce number.
```

Comment:
Clear instructions. But how does this actually work? Does a mapping need to be generated for each EAS ID to an odoo field? Can the mapping be created by a script?
In this example the Dutch Chamber Of Commerce number is added by oca module partner_coc (from V15 and up in odoo core). So could it be mapped automatically?

There is only one field for the mapping. But according to the standard more options are available and allowed.
Those are in this current version not included in the output.
Some of them could be constructed automatically like dutch `VAT` number, `9944`
(If partner country code is NL then VAT number code is  `9944`. If partner country code is DE then VAT number code is `9930`.

Reply:
I think these could be automatically mapped, but since that is a long and tedious work, it fell out of the scope of the budget for this project. I'll add it to the roadmap.

For the `99XX` codes, I'm not sure these are actually allowed, because when I used one I got this validation error:

![image](https://user-images.githubusercontent.com/1466356/194902744-83248832-dc21-4354-8f6a-dae4e65323bf.png)

Seems only the lower codes are allowed in the standard.